File: test/benchmark/map_numeric.rb
start = Time.now

map = Hash.new

for i in (1..2000000)
  map[i] = i
end

sum = 0
for i in (1..2000000)
  sum = sum + map[i]
end
puts sum

for i in (1..2000000)
  map.delete(i)
end

puts "elapsed: " + (Time.now - start).to_s

File: test/benchmark/map_numeric.py
from __future__ import print_function

import time

start = time.process_time()

map = {}

for i in range(1, 2000001):
  map[i] = i

sum = 0
for i in range(1, 2000001):
  sum = sum + map[i]
print(sum)

for i in range(1, 2000001):
  del map[i]

print("elapsed: " + str(time.process_time() - start))